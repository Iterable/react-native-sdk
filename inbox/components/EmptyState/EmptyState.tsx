"use strict"
import React, { Component } from "react"
import { View, Text, StyleSheet } from "react-native"

const EmptyState = () => {
   const title = "No saved messages"
   const subTitle = "Check again later!"

   return(
      <View style={styles.container}>
         <Text testID="title" style={styles.title}>
            {title}    
         </Text>
         <Text testID="subtitle" style={styles.subTitle}>
            {subTitle}    
         </Text>
      </View>     
   )
}

const styles = StyleSheet.create({
    container: {
       height: '100%',
       backgroundColor: 'whitesmoke', 
       flexDirection: 'column',
       justifyContent: 'flex-start',
       alignItems: 'center'
    },
 
    title: {
       fontWeight: 'bold',
       fontSize: 20,
       paddingTop: 250,
       paddingBottom: 25
    },

    subTitle: {
       fontSize: 15,
       color: 'grey'
    }
 })

export default EmptyState;