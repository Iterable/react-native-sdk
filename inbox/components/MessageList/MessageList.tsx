'use strict'
import React, { ReactElement } from 'react'
import { ScrollView, Text, StyleSheet } from 'react-native'
import MessageCell from '../MessageCell/MessageCell'

type MessageListProps = {
   messages: Array<any>  
}

const MessageList = ({ messages }: MessageListProps) => {
   function flaggedMessage(message: { [key: string]: any }, index: number) {
      return (index === messages.length - 1) ?
         <MessageCell key={index} message={message} last={true} /> :
         <MessageCell key={index} message={message} last={false} />
   }

   function displayMessages() {
      return messages.map((message, index) => {
         return flaggedMessage(message, index)
      })
   }

   return(
      <ScrollView style={styles.container}>
         {displayMessages()}
      </ScrollView>   
   )
}

const styles = StyleSheet.create({
   container: {
      flex: 1,
      // backgroundColor: '',
      // width: '100%'
   }
})

export default MessageList;