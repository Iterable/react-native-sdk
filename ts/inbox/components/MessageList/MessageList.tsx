'use strict'
import React, { Component } from 'react'
import { ScrollView, Text, StyleSheet } from 'react-native'
import MessageCell from '@iterable/react-native-sdk/js/Inbox/components/MessageCell/MessageCell'

function MessageList() {
   return(
      <ScrollView style={styles.container}>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>
         <MessageCell lastMessage={false}></MessageCell>           
         <MessageCell lastMessage={true}></MessageCell>
      </ScrollView>   
   )
}

const styles = StyleSheet.create({
   container: {
      flex: 1,
      // backgroundColor: '',
      // width: '100%'
   }
})

export default MessageList;