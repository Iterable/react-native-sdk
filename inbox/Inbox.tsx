'use strict'
import React, { Component, useState } from 'react'
import { Text, View, StyleSheet } from 'react-native'
import MessageList from './components/MessageList/MessageList'
import EmptyState from './components/EmptyState/EmptyState'
import sampleMessages from './sampleMessageData.js' 

export const showMessageList = (messages) => {
   return messages.length ? <MessageList messages={messages}></MessageList> : <EmptyState></EmptyState>
}

const Inbox = () => {
   const message = "Inbox";
   const [messages, setMessages] = useState(sampleMessages);

   return(
      <View style={styles.container}>
         <Text style={styles.headline}>
            {message}
         </Text>
         {showMessageList(messages)}
      </View>  
   )
}

const styles = StyleSheet.create({
   container: {
      height: '100%',
      backgroundColor: 'whitesmoke', 
      flexDirection: 'column',
      justifyContent: 'flex-start',
      marginTop: 50
   },

   headline: {
      fontWeight: 'bold' ,
      fontSize: 40,
      width: '100%',
      paddingTop: 10,
      paddingBottom: 10,
      paddingLeft: 15,
      backgroundColor: 'whitesmoke'
   }
})

export default Inbox;