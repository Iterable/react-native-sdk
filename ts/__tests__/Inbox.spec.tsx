import React from 'react'
import Inbox, {showMessageList} from '../../inbox/Inbox'
import MessageList from '../../inbox/components/MessageList/MessageList'
import EmptyState from '../../inbox/components/EmptyState/EmptyState'
import sampleMessages from '../../inbox/sampleMessageData.js' 

import renderer from 'react-test-renderer'

beforeEach(() => {
   jest.clearAllMocks()
})

describe('Testing Inbox Component functionality', () => {
   it('Inbox renders correctly', () => {
      let component = renderer.create(
         <Inbox/>
      ).toJSON()
      expect(component).toMatchSnapshot()  
   })
})

describe('Testing showMessageList function', () => {
   it('renders message list when messages are present', () => {
      const messages = sampleMessages
      const result = <MessageList messages={sampleMessages}/>
      expect(showMessageList(messages)).toEqual(result)
   })
   
   it('renders empty state when there are no messages', () => {
      const messages = []
      const result = <EmptyState/>
      expect(showMessageList(messages)).toEqual(result)
   })
})