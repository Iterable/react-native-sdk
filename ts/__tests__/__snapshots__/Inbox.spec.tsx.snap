// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Inbox Component functionality Inbox renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "whitesmoke",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "flex-start",
      "marginTop": 50,
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "whitesmoke",
        "fontSize": 40,
        "fontWeight": "bold",
        "paddingBottom": 10,
        "paddingLeft": 15,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  >
    Inbox
  </Text>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            CATS FOR SALE!! LIMITED TIME!!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            All this week!!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "paddingLeft": 30,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            THIS WEEKEND ONLY
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Trucks Trucks Trucks
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Iterable wants you!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Hackathon going on Monday
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "paddingLeft": 30,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Happy Birthday Buddy
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            All the cake for you!?
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Dog Moms Unite!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Show up at the dog park
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Dog Moms Unite!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Show up at the dog park
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Dog Moms Unite!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Show up at the dog park
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Dog Moms Unite!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Show up at the dog park
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Dog Moms Unite!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Show up at the dog park
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
            "borderBottomColor": "lightgray",
            "borderBottomWidth": 1,
            "borderStyle": "solid",
            "borderTopColor": "lightgray",
            "borderTopWidth": 1,
            "borderWidth": 0,
            "flexDirection": "row",
            "marginBottom": 70,
            "paddingBottom": 10,
            "paddingTop": 10,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "column",
              "height": "100%",
              "justifyContent": "flex-start",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "blue",
                "flexDirection": "row",
                "fontSize": 60,
                "justifyContent": "flex-start",
              }
            }
          >
            •
          </Text>
        </View>
        <View
          style={
            Object {
              "paddingLeft": 5,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 22,
                "paddingBottom": 10,
              }
            }
          >
            Dog Moms Unite!
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 15,
                "paddingBottom": 10,
              }
            }
          >
            Show up at the dog park
          </Text>
          <Text
            style={
              Object {
                "color": "lightgray",
                "fontSize": 12,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
